manifests/server/config.pp

```diff
@@ -222,21 +226,15 @@
             require => File['systemd-conf-dir'],
         ;
 
-        # Remove old unit file to avoid service failures
+        # Remove old unit file to avoid conflicts
         'old-systemd-override':
             ensure => absent,
             path   => "/etc/systemd/system/${service_name}.service",
         ;
       }
-
-      exec { 'restart-systemd':
-        command     => 'systemctl daemon-reload',
-        refreshonly => true,
-        path        => '/bin:/usr/bin:/usr/local/bin',
-      }
     }
 
-    if $facts['os']['release']['major'] == '7' or ($facts['os']['name'] == 'Fedora' and $facts['os']['release']['major'] <= '31') {
+    if $facts['os']['name'] == 'Fedora' and $facts['os']['release']['major'] <= '31' {
```

Comment:
Old Fedora also works well with drop ins. I think it's safe to drop this part altogether.

Reply:
That makes sense and TBH do we even *care* about old Fedora releases?  Anything older than F32 is now EOL and anybody running Fedora in production would be advised to upgrade.

Reply:
Exactly. If EL7 supports it, that means ancient Fedora supports it. Fedora isn't even listed in metadata.json so technically there is no guarantee at all that it works.

Reply:
I've updated the conditional OS checks using the changes suggested and also removed unit tests for ancient Fedora releases.